src/main/scala/is/hail/expr/HailRep.scala

```diff
@@ -62,20 +63,20 @@ trait HailRepFunctions {
     def typ = TGenotype
   }
 
-  implicit object variantHr extends HailRep[Variant] {
-    def typ = TVariant
+  implicit def variantHr[T](implicit hrt: HailRep[T]) = new HailRep[Variant] {
```

Comment:
This shouldn't have a HailRep as a parameter, since it isn't parameterized by a Type but by a GenomeReference.  There should be a new class that mirrors TVariable but isn't in the type hierarchy.  Something like GRVariable which matches a reference with unification.  It should be a GenomeReference so you can construct a TVariant from one.